.github/workflows/documentation.yml

```diff
@@ -0,0 +1,48 @@
+name: Documentation
+
+on:
+  push:
+    branches: main
+    tags: '*'
+  workflow_dispatch:  # Allow manually triggering the workflow
+
+concurrency:
+  group: >-
+    ${{ github.workflow }}-${{ github.ref_type }}-
+    ${{ github.event.pull_request.number || github.sha }}
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with: {python-version: "3.11"}
+      - name: Build Docs
+        run: |
+          pip install tox
+          tox -e docs-py311
+      - name: Upload docs artifact
+        uses: actions/upload-pages-artifact@v1
+        with:
+          path: 'build/docs'
+
+  deploy:
+    needs: build
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      pages: write
+      id-token: write
+    environment:
+      name: github-pages
+      url: ${{ steps.deployment.outputs.page_url }}
+    steps:
+      - name: Deploy to GitHub Pages
+        id: deployment
+        uses: actions/deploy-pages@v1
```

Comment:
is this a github action or from a third party? I think not all third party actions are allowed, so just asking

Reply:
I do not know, taken from Foundry Dev Tools, so I assume we should be allowed to use it?

Reply:
I guess so, seems like its a github aciton, but settings between repos could be different, will only find out `on-test`